lib/pry-doc/pry_ext/show_source_with_c_internals.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "show_source_with_c_internals/code_fetcher"
+
+module Pry::CInternals
+  class ShowSourceWithCInternals < Pry::Command::ShowSource
+    def options(opt)
+      super(opt)
+      opt.on :c, "c-source", "Show source of a C symbol in MRI"
+    end
+
+    def show_c_source
+      if opts.present?(:all)
+        result, file = CodeFetcher.new(line_number_style).fetch_all_definitions(obj_name)
+      else
+        result, file = CodeFetcher.new(line_number_style).fetch_first_definition(obj_name)
+      end
+      if result
+        set_file_and_dir_locals(file)
+        _pry_.pager.page result
+      else
+        raise Pry::CommandError, no_definition_message
+      end
+    end
+
+    def process
+      if opts.present?(:c) && !Pry.config.skip_mri_source
+        show_c_source
+        return
+      else
+        super
+      end
+    rescue Pry::CommandError
+      raise if Pry.config.skip_mri_source
+      show_c_source
+    end
+
+    private
+
+    # We can number lines with their actual line numbers
+    # or starting with 1 (base-one)
+    def line_number_style
+      style = if opts.present?(:'base-one')
+                :'base-one'
+              elsif opts.present?(:'line-numbers')
+                :'line-numbers'
+              else
+                nil
+              end
+
+      { line_number_style: style }
```

Comment:
I would expect a method named `line_number_style` to return `style`, not a Hash. If it's a Hash, then `code_fetcher_params` sounds more appropriate. I think this method can return `style`, then on line 11 we can write:

```rb
code_fetcher = CodeFetcher.new(line_number_style: line_number_style)
result, file =
  if opts.present?(:all)
    code_fetcher.fetch_all_definitions(obj_name)
  else
    code_fetcher.fetch_first_definition(obj_name)
  end
```

Reply:
agree :+1: